### ISSUE TITLE
5.0.0rc10: `autogen.pl --force` failed

### ISSUE THREAD

**Issue by mkurnosov** [open-mpi/ompi#11406]
### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

5.0.0rc10

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

source tarball (openmpi-5.0.0rc10.tar.bz2)

### Please describe the system on which you are running

* Operating system/version: 5.4.0-135-generic (Ubuntu 20.04.5 LTS)
* Computer hardware: x86_64
* Network type: IB

-----------------------------

## Details of the problem

`./autogen.pl --force` failed.
File `3rd-party/autogen.subdirs` contains a path to a non-existent directory `romio341`

```shell
./autogen.pl --force
Open MPI autogen (buckle up!)

1. Checking tool versions

   Searching for autoconf
     Found autoconf version 2.69; checking version...
       Found version component 2 -- need 2
       Found version component 69 -- need 69
       Found version component  -- need 0

     ==> ACCEPTED
   Searching for libtoolize
     Found libtoolize version 2.4.6; checking version...
       Found version component 2 -- need 2
       Found version component 4 -- need 4
       Found version component 6 -- need 2

     ==> ACCEPTED
   Searching for automake
     Found automake version 1.16.1; checking version...
       Found version component 1 -- need 1
       Found version component 16 -- need 13
     ==> ACCEPTED

2. Checking for git submodules

3. Running template-generating scripts

...

7. Processing autogen.subdirs directories
=== Processing subdir: 3rd-party/openpmix/
--- Found autogen.pl; running...
...

=== Processing subdir: 3rd-party/prrte/
--- Found autogen.pl; running...
...

=== Processing subdir: 3rd-party/romio341
Could not change to 3rd-party/romio341
```

**Comment by jjhursey** [comment]
I believe that this is related to https://github.com/open-mpi/ompi/issues/11364, which is still unresolved.

**Comment by jsquyres** [comment]
@mkurnosov This is now fixed on `main`, and will be fixed in `v5.0.x` by #11453.
